### ISSUE TITLE
Error in computing `MulticlassRecall` macro average when `ignore_index` is specified

### ISSUE THREAD

**Issue by JeroenMandersloot** [Lightning-AI/torchmetrics#2441]
## 🐛 Bug

In `MulticlassRecall` the `ignore_index` argument leads to incorrect behaviour when computing the macro average. It seems as if the metric still divides by the total number of classes rather than ignoring the `ignore_index` class (while treating the `ignore_index` class as having a score of `0`).

### To Reproduce

```python
import torch
from torchmetrics.classification.precision_recall import MulticlassRecall

metric = MulticlassRecall(num_classes=2, ignore_index=0, average="macro")
y_true = torch.tensor([0, 0, 1, 1])
y_pred = torch.tensor([
    [0.9, 0.1],  # Correct
    [0.9, 0.1],  # Correct
    [0.9, 0.1],  # Incorrect
    [0.1, 0.9],  # Correct
])
metric.update(y_pred, y_true)
print(metric.compute())  # Prints 0.25, but I would expect the result to be 0.5
```

### Expected behavior

I would expect the recall score to be `0.5`.

In the toy example above I have two classes (0 and 1) and two instances of each class. I want to ignore class 0 altogether (hence `ignore_index=0`). The first prediction for class 1 (the third prediction overall) is incorrect (it predicts class 0 with confidence 0.9 instead), but the other prediction is correct. Therefore, I would expect the recall score to be `0.5`. 

However, the actual result is `0.25`. It looks as if this metric is still dividing by the total number of classes when computing the macro average instead of ignoring the `ignore_index` class (while treating the `ignore_index` class as having a score of `0`).

(Notice that the predictions for class 0 are in fact correct. If I leave out the `ignore_index=0` altogether, I get an expected result of `0.75`.)

### Environment

- TorchMetrics version: 1.3.1 (installed via pip)
- Python version: 3.10
- PyTorch version: 2.2.1
- OS: Ubuntu 22.04

**Comment by ding3820** [comment]
This issue has been there for a while. After v0.9.3, `average="macro"` does not work with `ignore_index`.

**Comment by Borda** [comment]
> This issue has been there for a while. After v0.9.3, `average="macro"` does not work with `ignore_index`.

thank you for the adjustment, would yo be interested in sending a PR?

**Comment by rittik9** [comment]
Pls assign it to me @Borda

**Comment by Borda** [comment]
> Pls assign it to me @Borda

cc: @SkafteNicki

**Comment by robmarkcole** [comment]
Any plans to address? Impacting torchgeo CC @adamjstewart
